src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Body,
+  Param,
+  Query,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { AllowedOrganization } from './entities/allowed-organization.entity';
+import { OrganizationStatus } from 'src/common/constants';
+import { GithubInstallationService } from './github.installation.service';
+import { ApiTags } from '@nestjs/swagger';
+import { CreateOrgDto } from './dto/create.org.dto';
+
+@ApiTags('Github-Installation')
```

Comment:
PullProofAI

Consider adding input validation for `CreateOrgDto` in the `create` method to ensure data integrity.
```typescript
@Post()
async create(
  @Body() createOrgDto: CreateOrgDto,
): Promise<AllowedOrganization> {
  // Validate input data before processing
  const errors = await validate(createOrgDto);
  if (errors.length > 0) {
    throw new HttpException('Validation failed', HttpStatus.BAD_REQUEST);
  }

  try {
    return this.githubInstallationService.create(createOrgDto);
  } catch (error) {
    if (error.status) {
      throw new HttpException(error.message, error.status);
    } else {
      throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
    }
  }
}
```
Adding input validation ensures that the data passed to the `create` method is valid before processing it further.



<!-- This is an auto-generated comment by PullProofAI -->